runtime/ftplugin/help_toc.vim

```diff
@@ -0,0 +1,66 @@
+if !exists('g:no_plugin_maps')
+  function! s:show_toc() abort
+    if !exists('b:qf_toc')
```

Comment:
> Can the qf title be used?

I guess `bufname()` could be attached to it, but that can be a little long and not exactly informative (in a useful way) since man and help should only have one window, and therefore one TOC.

Since the method now is to open the TOC when the key is pressed, I could just set a variable in the quickfix buffer that has the help/man's filename in it to track this.

> Unreliable because the user could edit it?

No.  A bit of a rant that's not directly related to this PR: It's just unreliable as a source of information regarding its relationship to another window.  Quickfix is meant to be programmatically filled, but setting the list is a one-way deal that strips away unknown dict keys that could've been used by a script to manage the list.  State info can't be attached until the quickfix window is opened, so, you can't just set the list then wait until the user has displayed it to script it further.

Also, the addition of the `{what}` argument to the quickfix functions is half-assed.

> The worst that happens is the TOC is recalculated

The worst case is bad UX.  I wanted to avoid the recalculation because it can be slow, and it also resets the quickfix selection index.  A `<M-]>` when the TOC is already opened for the current file should just move the cursor back to the TOC for browsing/searching.